client/src/pages/ConfirmSignupPage/ConfirmSignupPage.js

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import './ConfirmSignupPage.scss';
+import { connect } from 'react-redux';
+import { withRouter, Redirect } from 'react-router-dom';
+import * as QueryString from 'query-string';
+import { Button, Card } from 'react-bootstrap';
+import { confirmSignup, resendConfirmation } from '../../actions/authentication';
+import Spinner from '../../components/Spinner/Spinner';
+
+class ConfirmSignupPage extends React.Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            redirect: false,
+            resend: false
+        };
+    }
+
+    componentDidMount() {
+        this.initialize();
+    }
+
+    initialize() {
+        console.log('init');
+        const { dispatch, location: { search }, session } = this.props;
+        const params = search && QueryString.parse(search);
+
+        const token = params && params.token;
+        const resend = params && params.resend;
+
+        if (resend) {
+            console.log('resend', session.user);
+            if (session.user) {
+                return this.resendConfirmation();
+            } else {
+                return this.setState({resend: true});
+            }
+        }
+
+        if (token) {
+            return dispatch(confirmSignup(token));
+        }
+
+        // How did you get here you little rascal?
+        return this.setState({redirect: '/'});
+    }
+
+    static mapStateToProps(state) {
+        return {
+            confirmSignup: state.confirmSignup,
+            resendConfirmation: state.resendConfirmation,
+            session: state.session
+        };
+    }
+
+    async beginResendConfirmation() {
+        await this.setState({redirect: '/signup/confirm?resend=true'});
+        this.initialize();
+    }
+
+    resendConfirmation() {
+        const { dispatch, session: { user } } = this.props;
+        dispatch(resendConfirmation(user));
+    }
+
+    render() {
+        const { confirmSignup, resendConfirmation, session } = this.props;
+        const { resend, redirect } = this.state;
+
+        if (redirect) {
+            return (<Redirect to={redirect} push/>)
```

Comment:
What does the `push` prop do?

Reply:
https://reacttraining.com/react-router/web/api/Redirect/push-bool